model/task/db.go

```diff
@@ -302,9 +302,11 @@ func ByTimeRun(startTime, endTime time.Time) db.Q {
 // ByTimeStartedAndFailed returns all failed tasks that started between 2 given times
 func ByTimeStartedAndFailed(startTime, endTime time.Time, commandTypes []string) db.Q {
 	query := bson.M{
-		StartTimeKey: bson.M{"$lte": endTime},
-		StartTimeKey: bson.M{"$gte": startTime},
-		StatusKey:    evergreen.TaskFailed,
+		"$and": []bson.M{
+			{StartTimeKey: bson.M{"$lte": endTime}},
+			{StartTimeKey: bson.M{"$gte": startTime}},
```

Comment:
maybe we could add a third condition to check start_time is set? (for when all the tasks are terminated by the task timeout monitor)

Reply:
Well this query implicitly checks that start_time is set (i.e. tasks terminated by the task timeout monitor won't be picked up). Although I did some DB digging and while there are many instances of start_time being unset for versions in staging (as I pointed out to you earlier), I don't see this in prod, so I don't think we need to case specially for it.

Reply:
Well this query implicitly checks that start_time is set (i.e. tasks terminated by the task timeout monitor won't be picked up). Despite what I said earlier these cases probably should be picked up, I think maybe if the task finished during the time range. I added a case to the queries to account for this.